### PR TITLE
Use init.sh bash script as main starting point

### PR DESCRIPTION
**Pull request by electrocucaracha**
This PR centralizes the`init.sh` script usage. Terraform (e2e CI process) and Vagrant (local) provisioning process will be using the same script. This will provide the same developer experience for the distinct provisioning process.

### CODE REVIEW

**Comment by electrocucaracha** [comment]
@johnbelamaric @vjayaramrh this PR fixes the e2e CI task. Sandbox environments are provisioned in ~15 mins, after that tests are run. Once the e2e job is completed, you can verify its [log](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/93/e2e/1668815328945115136).

**Comment by liamfallon** [comment]
We'll test this PR on an openstack VM today.

File: e2e/provision/gce_init.sh [hunk 1]
```diff
@@ -28,32 +28,39 @@ RUN_E2E=${NEPHIO_RUN_E2E:-$(get_metadata nephio-run-e2e "false")}
 REPO=${NEPHIO_REPO:-$(get_metadata nephio-test-infra-repo "https://github.com/nephio-project/test-infra.git")}
 BRANCH=${NEPHIO_BRANCH:-$(get_metadata nephio-test-infra-branch "main")}
 NEPHIO_USER=${NEPHIO_USER:-ubuntu}
+HOME=${HOME:-/home/$NEPHIO_USER}
+NEPHIO_REPO=${NEPHIO_REPO:-$HOME/test-infra}
```
Comment (johnbelamaric) [inline e2e/provision/gce_init.sh:32]: NEPHIO_REPO is expected to be a Git URL when passed from the command line. That is, the script supports calling it with

NEPHIO_REPO=https://github.com/my-user/nephio-test-infra.git ./gce_init.sh

I don't think we actually *do* that anywhere. But this use of NEPHIO_REPO seems to be the local clone directory, not the upstream repository.

Comment (johnbelamaric) [reply e2e/provision/gce_init.sh:32]: I would suggest naming this something like "REPO_DIR" or "LOCAL_REPO" or something.

Comment (electrocucaracha) [reply e2e/provision/gce_init.sh:32]: Good catch, I didn't notice that was used previously for other purpose.

**Comment by liamfallon** [comment]
Trying this now on an openstack VM:

```
cat gce_init.sh | sudo NEPHIO_USER=ubuntu DEBUG=true bash
false, r1, false, https://github.com/nephio-project/test-infra.git, main, ubuntu
fatal: could not create work tree dir '/root/test-infra': Permission denied
```
This happens because $HOME has the value _/root_

I copied the _gce_init.sh_  from your PR onto the VM. I can get past this by passing in HOME as /home/ubuntu but it would probably be better to change line 31 of gec_init.sh as follows to avoid confusion with the usage of the system varialble HOME a
```
from:
HOME=${HOME:-/home/$NEPHIO_USER}
to:
HOME=${NEPHIO_HOME:-/home/$NEPHIO_USER}

Then if NEPHIO_HOME is set, it is used otherwise it defaults to /home/$NEPHIO_USER and the install command above will work

File: e2e/provision/gce_install_sandbox.sh [hunk 2]
```diff
@@ -68,20 +65,15 @@ cat "$HOME/.ssh/id_rsa.pub" >>"$HOME/.ssh/authorized_keys"
...
-    [[ -d "$HOME/gtp5g" ]] || git clone --depth 1 -b v0.6.8 https://github.com/free5gc/gtp5g.git "$HOME/gtp5g"
-
-    pushd "$HOME/gtp5g" >/dev/null
-    GCC=gcc
-    if [[ $(uname -v) == *22.04.*-Ubuntu* ]]; then
-        GCC=gcc-12
-    fi
-    if ! command -v $GCC >/dev/null; then
-        sudo apt-get -y install $GCC
+if [[ ${LOAD_GTP5G_MODULE:-false} == "true" ]] && ! lsmod | grep -q gtp5g; then
```
Comment (electrocucaracha) [inline e2e/provision/gce_install_sandbox.sh:68]: Maybe I have to enable by default
```suggestion
if [[ ${LOAD_GTP5G_MODULE:-true} == "true" ]] && ! lsmod | grep -q gtp5g; then
```

**Comment by liamfallon** [comment]
Just ran the management cluster install using this PR on a VM running on Openstack and it worked fine.

**Comment by electrocucaracha** [comment]
> Trying this now on an openstack VM:
> 
> ```
> cat gce_init.sh | sudo NEPHIO_USER=ubuntu DEBUG=true bash
> false, r1, false, https://github.com/nephio-project/test-infra.git, main, ubuntu
> fatal: could not create work tree dir '/root/test-infra': Permission denied
> ```
> 
> This happens because $HOME has the value _/root_
> 
> I copied the _gce_init.sh_ from your PR onto the VM. I can get past this by passing in HOME as /home/ubuntu but it would probably be better to change line 31 of gec_init.sh as follows to avoid confusion with the usage of the system varialble HOME a
> 
> ```
> from:
> HOME=${HOME:-/home/$NEPHIO_USER}
> to:
> HOME=${NEPHIO_HOME:-/home/$NEPHIO_USER}
> 
> Then if NEPHIO_HOME is set, it is used otherwise it defaults to /home/$NEPHIO_USER and the install command above will work
> ```

That's right, the `NEPHIO_REPO_DIR` is going to give us some flexibility to reuse code or to clone it.

**Comment by radoslawc** [comment]
I've just ran it in vagrant in my lab and it worked fine.

**Comment by electrocucaracha** [comment]
> /retest

The failures on this are not caused by this change, those are timeouts raised by the testing scripts

**Comment by electrocucaracha** [comment]
The [timeout error getting here](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/93/e2e/1669062975417225216#1:build-log.txt%3A51485) is not related with this PR, this only exposes the current state.

**Pull request merged**